src/Stratis.Bitcoin.Features.BlockStore/PrunedBlockRepository.cs

```diff
@@ -0,0 +1,179 @@
+
+            while (startFromHeader.Previous != null && startFromHeader != endAtHeader)
+            {
+                toDelete.Add(startFromHeader);
+                startFromHeader = startFromHeader.Previous;
+            }
+
+            await this.blockRepository.DeleteBlocksAsync(toDelete.Select(cb => cb.HashBlock).ToList()).ConfigureAwait(false);
+
+            this.UpdatePrunedTip(consensusTip.GetAncestor(upperHeight));
+        }
+
+        private HashHeightPair LoadPrunedTip(DBreeze.Transactions.Transaction dbreezeTransaction)
+        {
+            if (this.PrunedTip == null)
+            {
+                dbreezeTransaction.ValuesLazyLoadingIsOn = false;
+
+                Row<byte[], HashHeightPair> row = dbreezeTransaction.Select<byte[], HashHeightPair>(BlockRepository.CommonTableName, prunedTipKey);
+                if (row.Exists)
+                    this.PrunedTip = row.Value;
+
+                dbreezeTransaction.ValuesLazyLoadingIsOn = true;
+            }
+
+            return this.PrunedTip;
+        }
+
+        /// <summary>
+        /// Compacts the block and transaction database by recreating the tables without the deleted references.
+        /// </summary>
+        private void CompactDataBase()
+        {
+            Task task = Task.Run(() =>
+            {
+                using (DBreeze.Transactions.Transaction dbreezeTransaction = this.blockRepository.DBreeze.GetTransaction())
+                {
+                    dbreezeTransaction.SynchronizeTables(BlockRepository.BlockTableName, BlockRepository.TransactionTableName);
+
+                    var tempBlocks = dbreezeTransaction.SelectDictionary<byte[], Block>(BlockRepository.BlockTableName);
+
+                    if (tempBlocks.Count != 0)
+                    {
+                        this.logger.LogInformation($"{tempBlocks.Count} blocks will be copied to the pruned table.");
+
+                        dbreezeTransaction.RemoveAllKeys(BlockRepository.BlockTableName, true);
+                        dbreezeTransaction.InsertDictionary(BlockRepository.BlockTableName, tempBlocks, false);
+
+                        var tempTransactions = dbreezeTransaction.SelectDictionary<byte[], Block>(BlockRepository.TransactionTableName);
+                        if (tempTransactions.Count != 0)
+                        {
+                            this.logger.LogInformation($"{tempTransactions.Count} transactions will be copied to the pruned table.");
+                            dbreezeTransaction.RemoveAllKeys(BlockRepository.TransactionTableName, true);
+                            dbreezeTransaction.InsertDictionary(BlockRepository.TransactionTableName, tempTransactions, false);
+                        }
+
+                        // Save the hash and height of where the node was pruned up to.
+                        dbreezeTransaction.Insert(BlockRepository.CommonTableName, prunedTipKey, this.PrunedTip);
```

Comment:
This is compacting why do we need to save the pruned tip?

Reply:
The Pruning Service uses this to determine if the node has been pruned or not and from what height its need to prune.